tox.ini

```
@@ -1,37 +1,75 @@
 [tox]
 description = Default tox environments list
 envlist =
-    style,{py38,py39,py310,py311,py312}{,-coverage},doc
+    code-style
+    doc-style
+    {py39,py310,py311,py312}{,-coverage}
+    doc-{links,html}
+    docker-run-{linux_host,windows_host}
 skip_missing_interpreters = true
 isolated_build = true
 isolated_build_env = build
 
 [testenv]
 description = Checks for project unit tests and coverage (if desired)
 basepython =
-    py39: python3.9
-    py310: python3.10
-    py311: python3.11
-    py312: python3.12
+    tests39: python3.9
+    tests310: python3.10
+    tests311: python3.11
+    tests312: python3.12
     py: python3
-    {style,reformat,doc}: python3
+    {code-style,doc-links,doc-html, tests}: python3
 setenv =
     PYTHONUNBUFFERED = yes
     coverage: PYTEST_EXTRA_ARGS = --cov=ansys.mapdl --cov-report=term --cov-report=xml:.cov/xml --cov-report=html:.cov/html
 extras = tests
 commands =
     pytest {env:PYTEST_MARKERS:} {env:PYTEST_EXTRA_ARGS:} {posargs:-vv}
 
-[testenv:style]
+[testenv:code-style]
 description = Checks project code style
+skip_install = true
 deps =
     pre-commit
 commands =
     pre-commit install
     pre-commit run --all-files --show-diff-on-failure
 
-[testenv:doc]
+[testenv:doc-style]
+description = Checks project documentation style
+skip_install = true
+allowlist_externals =
+    vale
+commands =
+    vale sync --config="{toxinidir}/doc/.vale.ini"
+    vale --config="{toxinidir}/doc/.vale.ini" "{toxinidir}/doc" 
+
+[testenv:doc-{links, html}]
 description = Check if documentation generates properly
+setenv =
+    links: BUILDER = linkcheck
+    html: BUILDER = html
+passenv =
+    BUILD_API
+    BUILD_EXAMPLES
 extras = doc
 commands =
-    sphinx-build -d "{toxworkdir}/doc_doctree" doc/source "{toxworkdir}/doc_out" --color -vW -bhtml
\ No newline at end of file
+    sphinx-build -d "{toxworkdir}/doc_doctree" doc/source "{toxinidir}/doc/_build/{env:BUILDER}" --color -vW -b {env:BUILDER} -j auto
+
+[testenv:docker-run-{linux_host}]
```

Let's have here also the `windows_host` environment just to make it consistent with other libraries:

```suggestion
[testenv:docker-run-{linux_host,windows_host}]
```

---

Remember that Windows users can run containers using Linux if using Docker desktop.